crates/sui-adapter/src/execution_engine.rs

```diff
@@ -37,11 +44,23 @@ use sui_types::{
 };
 use sui_types::{
     is_system_package, SUI_CLOCK_OBJECT_ID, SUI_CLOCK_OBJECT_SHARED_VERSION,
-    SUI_SYSTEM_STATE_OBJECT_SHARED_VERSION,
+    SUI_FRAMEWORK_OBJECT_ID, SUI_SYSTEM_STATE_OBJECT_SHARED_VERSION,
 };
 
 use sui_types::temporary_store::TemporaryStore;
 
+pub struct AdvanceEpochParams {
+    pub epoch: u64,
+    pub next_protocol_version: ProtocolVersion,
+    pub storage_charge: u64,
+    pub computation_charge: u64,
+    pub storage_rebate: u64,
+    pub storage_fund_reinvest_rate: u64,
+    pub reward_slashing_rate: u64,
+    pub epoch_start_timestamp_ms: u64,
+    pub new_system_state_version: u64,
+}
```

Comment:
Thinking about this a bit more...I wonder if we can simplify things greatly with programable txns. As in we can break the behemoth "advance_epoch" move call into more smaller parts. For example, if we know that protocol version isn't changing, why do we need to pass it in every single time and instead have that part be a separate step that only happens when we actually do a protocol upgrade. 

Anyway, these are just thoughts and not something we need to do for this PR.

cc @mystenmark

Reply:
i'm certainly open to simplifications, but in the specific case of protocol version, it seems like that would make it more complicated not less, since you'd have to check if the version has changed, and do something different if it has.